provider/cluster/hostname.go

```diff
@@ -4,92 +4,227 @@ import (
 	"context"
 	"fmt"
 	lifecycle "github.com/boz/go-lifecycle"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
 	dtypes "github.com/ovrclk/akash/x/deployment/types"
 	"github.com/pkg/errors"
 	"strings"
 	"sync"
 )
 
-type reserveRequest struct {
-	hostnames []string
-	result    chan<- error
-	doReserve bool
-	dID       dtypes.DeploymentID
-}
-
 type HostnameServiceClient interface {
-	ReserveHostnames(hostnames []string, did dtypes.DeploymentID) <-chan error
-	ReleaseHostnames(hostnames []string)
-	CanReserveHostnames(hostnames []string, did dtypes.DeploymentID) <-chan error
+	ReserveHostnames(hostnames []string, dID dtypes.DeploymentID) ReservationResult
+	ReleaseHostnames(dID dtypes.DeploymentID)
+	CanReserveHostnames(hostnames []string, ownerAddr sdktypes.Address) <-chan error
+	PrepareHostnamesForTransfer(hostnames []string, dID dtypes.DeploymentID) <- chan error
 }
 
 type SimpleHostnames struct {
 	Hostnames map[string]dtypes.DeploymentID
```

Comment:
can both of these be private?

Reply:
This type is used in tests to avoid the need to stop / start another goroutine in each test